CriteoPublisherSdk/Tests/IntegrationTests/CR_IntegrationsTestBase.m

```diff
@@ -39,9 +47,19 @@ - (void)waitForIdleState {
   [self.criteo.threadManager waiter_waitIdle];
 }
 
-- (void)tearDown {
-  [self waitForIdleState];
-  [super tearDown];
+#pragma mark - App Bidding helper
+
+- (void)enrichAdObject:(id)object
+             forAdUnit:(CRAdUnit *)adUnit
+           thenWaitFor:(void (^)(void))completionBlock {
+  [self cr_waitForExpectationBlock:^(XCTestExpectation *expectation) {
```

Comment:
You're doing all your assertions inside a block which make the tests a little bid harder to read (when the block is too big).
What do you think about mocking a synchronous call ? Either via XCTestExpectation or through `[self waitForIdleState]`.

Reply:
This is mostly a way to adapt all tests w/o being too intrusive. While it is maybe not the best one, that ease reviewing, and works with all scenarios, including I think ones that wasn't relying on idleState. I'll see if I can change this w/o too much rewrite / time investment.

Reply:
So your suggestion works in the end, took a bit more time that I was hoping, but anyway, that's better now, thank you :)